contracts/vault/CallWhitelistApprovals.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.11;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "./CallWhitelist.sol";
+import "../interfaces/IERC721Permit.sol";
+
+/**
+ * @title CallWhitelistApprovals
+ * @author Non-Fungible Technologies, Inc.
+ *
+ * Adds approvals functionality to CallWhitelist. Certain spenders
+ * can be approved for tokens on vaults, with the requisite ability
+ * to withdraw. Should not be used for tokens acting as collateral.
```

Comment:
Would it be useful to add a require statement that the token in the `Approve()` is not part of the loan terms?

Reply:
Well, I think that would break a nice separation of concerns - right now, the vault architecture knows nothing about loans at all - they're just vaults. So while it would be a good safety check I don't think it's the right architectural decision